source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -3707,6 +3710,40 @@ void Graph::searchNodePopup(bool cursor)
     }
 }
 
+bool Graph::isPinHovered()
+{
+    ed::PinId currentPin = ed::GetHoveredPin();
+    ed::PinId nullPin = 0;
+    return currentPin != nullPin;
+}
+
+void Graph::addPinPopup()
+{
+    // Add a floating popup to pin when hovered
+    if (isPinHovered())
+    {
+        ed::Suspend();
+        UiPinPtr pin = getPin(ed::GetHoveredPin());
+        std::string connected = "";
+        std::string value = "";
+        if (pin->_connected)
+        {
+            connected = "\nConnected to";
+            for (UiPinPtr connectedPins : pin->getConnections())
+            {
+                connected = connected + " " + connectedPins->_name + ",";
```

Comment:
@ethomare-ilm I'd recommend modifying this logic so that the comma is only added _between_ entries in the vector, and not after the last entry.